docker/android-system.sh

```diff
@@ -46,64 +79,474 @@ EOF
         python \
         python3 \
         xz-utils
-    purge_list+=(default-jre)
 
     curl --retry 3 -sSfL https://storage.googleapis.com/git-repo-downloads/repo -O
     chmod +x repo
+    python3 ./repo init -u https://android.googlesource.com/platform/manifest -b "${TAG}"
+
+    local tools=(
+        cat chmod chown cmp cp ctrlaltdel date df dmesg du hd id ifconfig
+        iftop insmod ioctl ionice kill ln log ls lsmod lsof lsusb md5 mkdir
+        mount mv nandread netstat notify printenv ps reboot renice rm rmdir
+        rmmod route schedtop sendevent setconsole setprop sleep smd start
+        stop sync top touch umount uptime vmstat watchprops wipe
+    )
+    if [[ "${ANDROID_SYSTEM_COMPLETE}" == "1" ]]; then
+        android_repo_complete
+    else
+        case "${MAJOR_VERSION}" in
+            5)
+                android_repo_v5
+                tools+=(dd getevent getprop grep newfs_msdos)
+                ;;
+            6)
+                android_repo_v6
+                ;;
+            7)
+                android_repo_v7
+                ;;
+            8)
+                android_repo_v8
+                ;;
+            9)
+                android_repo_v9
+                ;;
+            10)
+                android_repo_v10
+                ;;
+            11)
+                android_repo_v11
+                ;;
+            *)
+                echo "Currently unsupported Android version ${MAJOR_VERSION}." 1>&2
+                echo "Please submit a feature request if you need support." 1>&2
+                exit 1
+                ;;
+        esac
+    fi
+
+    build_android
+    install_android "${tools[@]}"
+
+    purge_packages
+
+    popd
+
+    rm -rf "${td}"
+    rm -rf "${PYTHON_TMPDIR}"
+    rm "${0}"
+}
+
+# java isn't required for the build, but the build expects to
+# find a java compiler. the supported android versions are:
+# https://source.android.com/docs/setup/start/older-versions
+#   Android 7: OpenJDK-8
+fake_java() {
+    local java_type=
+    local java_version=
+    local jre_info=
+    local build_info=
+
+    case "${MAJOR_VERSION}" in
+        5|6)
+            java_type=java
+            java_version=1.7.0
+            jre_info="IcedTea 2.6.9"
+            build_info="build 24.131-b00, mixed mode"
+            ;;
+        *)
+            java_type=openjdk
+            java_version=1.8.0_342
+            jre_info="build 1.8.0_342-8u342-b07-0ubuntu1~20.04-b07"
```

Comment:
Android 7+ changed the required Java version and forces an OpenJDK runtime environment. I've therefore changed it to use Java 1.8 and an OpenJDK runtime using the standard install on a Ubuntu-based system.